### PR TITLE
Clean-up configuration file options and associated code paths

### LINKED ISSUE #25: Clean-up unused code paths and associated configuration options
Currently there are various parts of the code which are on paths that are not hit when running with the set of provided example configuration files - for example loading from a JPEG dataset using `jpeg_dir`, loading additional metadata associated with NIfTI files using `load_metadata` and using this to partition the data files. We should decide which functionality should be retained and remove those features which are not of relevance here, or if they are to be retained provide example configuration files which can be used to test running with those features as part of #19. Creating some basic integration tests running the full workflow with the current example configuration and collecting coverage information may help with guiding what is a candidate for removal.

### PR DESCRIPTION
**Pull request by matt-graham**
Fixes #25 

Combines various clean-up related changes:

Removes various configuration file options which did not have any effect on code behaviour:

  * `new_model`
  * `apply_mask_in_input_space`
  * `include_mask_in_loader`
  * `discard_abnormally_small_niftis`
  * `nifti_standardise`
  * `shuffle_niftis`
  * `save_recons_to_mat`
  * `use_DDP`
  * `print_model`
  * `base_recons_on_train_loader`

Removes `sequence_type` configuration option, with only code paths corresponding to previous `"flair"` (rather than `"dwi"`) value retained.

Removes `use_nii_data` configuration option, with code paths corresponding to `use_nii_data == False` (JPEG dataset) removed.

Removes `likelihood` configuration option, with only code paths corresponding to previous `"Gaussian"` value retained.

Removes unused modules

  * `verydeepvae/data_tools/jpeg_dataset.py`
  * `verydeepvae/data_tools/additive_gaussian_noise_trans.py`
  * `verydeepvae/data_tools/binariser_augmentation.py`
  * `verydeepvae/data_tools/monaiwrapper_torchio_RandomGhosting.py`
  * `verydeepvae/data_tools/random_hemisphere_deleter.py`
  * `verydeepvae/data_tools/random_leave_random_cube.py`
  * `verydeepvae/data_tools/random_take_random_cube.py`
  * `verydeepvae/graph_components/identity_block.py`
  * `verydeepvae/graph_components/residual_block.py`
  * `verydeepvae/graph_components/sigmoid_channel_one_block.py`
  * `verydeepvae/graph_components/tanh_channel_one_block.py`

Removes unused functions / classes

  * `save_to_mat` and `compute_min_max_from_loader` in `verydeepvae/data_tools/data_handling.py`
  * `average_gradients`, `torch_nansafe_mean`, `dummy_context_mgr`, `kl_vars`, `kl_stds_then_log_vars`, `kl_vars_then_log_vars`, and `np_safe_divide`   from `verydeepvae/misc/misc.py`
  * `plot_slices`, `plot_3d_recons`, `save_niis`, `plot_2d_samples` and `plot_3d_samples` from `verydeepvae/misc/visuals.py`

Renames `verydeepvae/orchestration/training_script_vae_new.py` to `verydeepvae/orchestration/training.py` and renames `main` function in module `train_model`.

Renames `verydeepvae/graphs/vdeepvae_bottom_up_graph_translator.py` to `verydeepvae/graphs/bottom_up.py` and `verydeepvae/graphs/vdeepvae_top_down_graph_translator.py` to `verydeepvae/graphs/top_down.py`, and renames `Graph` classes in two modules to `BottomUpGraph` and `TopDownGraph` respectively.

Moves Matplotlib backend selection statement inside `train_model` function to avoid changing backed on import of package.

Changes `nii_target_shape` configuration file option (array of three integers) to `resolution` (a single integer) and updates logic in code accordingly.

Renames `latents_per_channel` configuration option to `latent_feature_maps_per_resolution`.

Replaces `use_sigmoid_output` and `use_tanh_output` boolean configuration file options to a single `output_activation_function` enumeration to avoid possibility of inconsistent states such as `use_sigmoid_output == True` and `use_tanh_output == True`.

Removes `use_abs_not_square` configuration option and associated code path. This corresponded to changing the likelihood from a Gaussian to a (non-standard parameterisation of a )Laplace distribution however this was not obvious from name of option and also conflicted with other comments / descriptions in code and documentation which assume Gaussian distributions.

References to `var` or `variance` in the configuration options are changed to `scale` to be consistent with convention of referring to the parameters of the Gaussian distribution as location and scale and to avoid ambiguities due to differing parameterisation in terms of standard deviation and variance in different parts of code.

Fixed error on plotting with configuration option `predict_x_scale` (previously `predict_x_var`) set to `False` due to `sample_std` array in `to_plot` list being `None` in this case causing error on subsequent attempt to index.

Increased coverage of boolean configuration file options in tests in `tests/test_script.py`.

### CODE REVIEW

**Comment by matt-graham** [comment]
After changes in this PR coverage report summary is

```
Name                                                             Stmts   Miss  Cover
------------------------------------------------------------------------------------
verydeepvae/__init__.py                                              0      0   100%
verydeepvae/data_tools/__init__.py                                   0      0   100%
verydeepvae/data_tools/clamp_by_percentile_augmentation.py          21      3    86%
verydeepvae/data_tools/crop_nii_by_given_amount.py                  13      6    54%
verydeepvae/data_tools/data_handling.py                             12      8    33%
verydeepvae/data_tools/data_transformations.py                     109      9    92%
verydeepvae/data_tools/random_anisotropiser_augmentation.py         33     12    64%
verydeepvae/data_tools/random_registered_3d_haircut.py              31      9    71%
verydeepvae/data_tools/torch_wrapper.py                             27      1    96%
verydeepvae/graph_components/__init__.py                             0      0   100%
verydeepvae/graph_components/conv_block.py                         179     38    79%
verydeepvae/graph_components/pooling_block.py                       21      1    95%
verydeepvae/graph_components/sigmoid_block.py                       11      5    55%
verydeepvae/graph_components/sigmoid_first_n_channels_block.py      12      0   100%
verydeepvae/graph_components/skip_block.py                          23     18    22%
verydeepvae/graph_components/tanh_block.py                          11      0   100%
verydeepvae/graph_components/tanh_first_n_channels_block.py         12      0   100%
verydeepvae/graph_components/top_down_block.py                     629    243    61%
verydeepvae/graph_components/unpooling_block.py                     19      3    84%
verydeepvae/graphs/__init__.py                                       0      0   100%
verydeepvae/graphs/bottom_up.py                                     43      9    79%
verydeepvae/graphs/top_down.py                                     183     69    62%
verydeepvae/misc/__init__.py                                         0      0   100%
verydeepvae/misc/environment.py                                    109     30    72%
verydeepvae/misc/misc.py                                           151     34    77%
verydeepvae/misc/visuals.py                                        173     68    61%
verydeepvae/orchestration/__init__.py                                0      0   100%
verydeepvae/orchestration/one_epoch.py                             209     28    87%
verydeepvae/orchestration/training.py                              467    127    73%
------------------------------------------------------------------------------------
TOTAL                                                             2498    721    71%
```

compared to [previous results here](https://github.com/high-dimensional/3d_very_deep_vae/issues/25#issuecomment-1131409538).

**Comment by r-gray** [comment]
This all looks absolutely fine to me!
